rmw_fastrtps_shared_cpp/src/namespace_prefix.cpp

```diff
@@ -31,8 +31,10 @@ const std::vector<std::string> _ros_prefixes =
 std::string
 _resolve_prefix(const std::string & name, const std::string & prefix)
 {
-  if (name.rfind(prefix, 0) == 0 && name.at(prefix.length()) == '/') {
-    return name.substr(prefix.length());
+  if (!name.empty()) {
```

Comment:
I may be missing something, but why don't you need this check in the other functions?

Reply:
Because the other functions don't use the [`std::string::at()`](https://en.cppreference.com/w/cpp/string/basic_string/at) method, which will throw if its argument is out of range.

Reply:
I think the [`std::string::rfind()`](https://en.cppreference.com/w/cpp/string/basic_string/rfind) function will return `npos` if the prefix is not found so the `at` function won't be used if the string is empty. Then the added check won't be needed.

Same below where the `at` function is used after a call to `rfind`.

Reply:
Ok, I think I get it now. But it leads to more questions actually. I think this logic needs to be updated and some more tests added. If the name or topic_name is "rr" or "rt" or "rq", these functions will throw. And lines 47 and 59 use std::string.at()

Reply:
All very good points. I've updated both implementation and tests in https://github.com/ros2/rmw_fastrtps/pull/388/commits/564379569c91890136ef6aa44f1a84bbb2237726